### PR TITLE
Make user docs for ISIS single-crystal diffraction reduction

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work**

Make user docs for ISIS single-crystal diffraction reduction.

In the course of this I found a couple of bugs/inconsistencies/user experience improvements:

1. Fixed typo in argument name in `WishSX.find_sx_peaks` (`ntube` -> `ntubes`)
2. Made argument name of `WishSX.remove_peaks_on_edge` to be consistent with `SXD.remove_peaks_on_edge`
3. Fix bug requiring string when ws object passed to `BaseSX.convert_ws_to_MD` (now accepts objects)
4. Add option to clear instrument calibration from peaks workspace in `SXD.undo_calibration`

Points 1-3 are covered in ISIS_WISHSingleCrystalReduction.py
Point 4 is covered in a unit test

**To test:**

(1) Run through code snippets in the docs (should not throw any errors)

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+        wish.load_isaw_ub(path.join(save_dir, f'{run}.mat'), tol=0.15, run=run)
+        # predict peaks
+        # main
+        wish.predict_peaks(MinDSpacing=0.75, WavelengthMin=0.85,
+                           ReflectionCondition='Body centred', run=run)
+        # satellite
+        wish.predict_peaks(peak_type=PEAK_TYPE.SATELLITE, run=run
+                           ModVector1="0.5,0,0", ModVector2="0,0.5,0", ModVector3="0,0,0.5",
+                           MaxOrder=1, CrossTerms=False, RequirePeaksOnDetector=True,
+                           ReflectionCondition='Body centred')
```
Comment (warunawickramasingha) [inline docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst:420]: I am getting this error at this line

`File "<unknown>", line 33
    ModVector1="0.5,0,0", ModVector2="0,0.5,0", ModVector3="0,0,0.5",
    ^
SyntaxError: invalid syntax`

**Review by warunawickramasingha — CHANGES_REQUESTED** [review]
I executed all the code blocks in the docs and found out a few locations with some errors which I have commented as above.

File: docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+  FindUBUsingFFT(PeaksWorkspace=peaks_ws , MinD=1, MaxD=10)
+  # force lattice parameters to be correct
+  SelectCellWithForm(PeaksWorkspace=peaks_ws, FormNumber=1, Apply=True, Tolerance=0.15)
+  OptimizeLatticeForCellType(PeaksWorkspace=peaks_ws, Apply=True, Tolerance=0.15)
+  IndexPeaks(PeaksWorkspace=peaks_ws, Tolerance=0.15, CommonUBForAll=False)
+  CalculateUMatrix(PeaksWorkspace=peaks_ws, a=5.6402, b=5.6402, c=5.6402,
+                   alpha=90, beta=90, gamma=90)
+
+  save_dir = r"/babylon/Public/UserName/"
+  detcal_path = sxd.calibrate_sxd_panels(wsname, pk_wsname, save_dir, tol=0.2)
```
[the remark this answers is not in the document]
Comment (RichardWaiteSTFC) [reply docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst:79]: Thanks, will do both of the above!

File: docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+                'NVacanciesMax': 2, 'NPixPerVacancyMin': 2, 'NTOFBinsMin': 2,
+                'UpdatePeakPosition': True, 'OptimiseMask': True,
+                'GetTOFWindowFromBackToBackParams': False,
+                'BackScatteringTOFResolution': 0.06, 'ThetaWidth': 1.5,
+                'ScaleThetaWidthByWavelength': True,
+                'OptimiseXWindowSize': True, 'ThresholdIoverSigma': 15}
+
+    # integrate and save each run
+    for run in runs:
+        skew_args = {**skew_args, 'OutputFile': path.join(save_dir, f"{run}_{peak_type.value}_int.pdf")}
```
[the remark this answers is not in the document]
Comment (RichardWaiteSTFC) [reply docs/source/techniques/ISIS_SingleCrystalDiffraction_Workflow.rst:172]: Thanks yes it should be `PEAK_TYPE.FOUND`

**Review by warunawickramasingha — APPROVED** [review]
Thank you for the updates and I am happy to approve!

**Pull request merged**